### ISSUE TITLE
User Defined Functors crash with inlined relations

### ISSUE THREAD

**Issue by b-scholz** [souffle-lang/souffle#1186]
There is an issue with user-defined functors and inlined relations. For example the following program crashes:

```
.functor hoo():number
  
.decl A(x:number) inline
A(1) :- 1 < 5.
          
.decl B(x:number) 
B(@hoo()) :- A(x).  
         
.output B
```

This issue was raised by @nevillegrech.

**Comment by mmcgr** [comment]
The crash happens at https://github.com/souffle-lang/souffle/blob/895f689b21d2726ff58a564e67bb76d65080ea3a/src/InlineRelationsTransformer.cpp#L632-L635

**Comment by mmcgr** [comment]
Fixed by #1188

**Comment by nevillegrech** [comment]
:+1:
